modules/__tests__/HashHistory-test.js

```diff
@@ -26,11 +26,18 @@ describe('hash history', function () {
   describePush(createHashHistory)
   describeReplaceState(createHashHistory)
   describeReplace(createHashHistory)
-  describePopState(createHashHistory)
   describeBasename(createHashHistory)
   describeQueries(createHashHistory)
 
-  if (supportsGoWithoutReloadUsingHash()) {
+  if (supportsHistory()) {
+    describePopState(createHashHistory)
+  } else {
+    describe.skip(null, function () {
+      describePopState(createHashHistory)
+    })
+  }
+
+  if (supportsHistory() && supportsGoWithoutReloadUsingHash()) {
```

Comment:
I'm -0 on doing the check here instead of in `supportsGoWithoutReloadUsingHash`. It could go in the latter, but I don't see much value in warning all IE9 users about this all the time, since it's not like there's a choice to use anything other than hash history there.